extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/BaseModel.java

```diff
@@ -110,6 +112,91 @@ public synchronized RouteDefinition getRouteDefinition(String id) {
         return null;
     }
 
+    @Override
+    public List<RouteTemplateDefinition> getRouteTemplateDefinitions() {
+        return routeTemplateDefinitions;
+    }
+
+    @Override
+    public RouteTemplateDefinition getRouteTemplateDefinition(String id) {
+        for (RouteTemplateDefinition route : routeTemplateDefinitions) {
+            if (route.idOrCreate(camelContext.adapt(ExtendedCamelContext.class).getNodeIdFactory()).equals(id)) {
```

Comment:
```suggestion
        final ... nodeIdFactory = camelContext.adapt(ExtendedCamelContext.class).getNodeIdFactory();
        for (RouteTemplateDefinition route : routeTemplateDefinitions) {
            if (route.idOrCreate(nodeIdFactory).equals(id)) {
```

It would be more effective to get the factory once before starting the loop.

Reply:
No problem, I'll refactor it